### PR TITLE
fix(TUP-26231)Relative path not working in Talend v721

### PR DESCRIPTION
**Pull request by jding-tlnd**
https://jira.talendforge.org/browse/TUP-26231

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/CodeGenerator.java [hunk 1]
```diff
@@ -353,8 +354,14 @@ public String generateContextCode(IContext designerContext) throws CodeGenerator
...
                             if (process instanceof IProcess2) {
-                                resourcePathForContext = resourceService.getResourcePathForContext(process,
-                                        contextPar.getValue());
+                                String value = contextPar.getValue();
+                                // for runtime
+                                if (!ProcessorUtilities.isExportConfig()) {
+                                    contextPar.setType(JavaTypesManager.FILE.getId());
+                                    resourcePathForContext = resourceService.getFullResourcePathForContext(process, value);
+                                } else {
+                                    resourcePathForContext = resourceService.getResourcePathForContext(process, value);
```
Comment (zwxue) [inline main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/CodeGenerator.java]: missing one caller of BigDataCodeGenerator, can move the isExportConfig to resourceService.getResourcePathForContext, no need add another api getFullResourcePathForContext

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/util/ResourceDependenciesUtil.java [hunk 2]
```diff
@@ -310,6 +310,21 @@ public static void copyToExtResourceFolder(JobResourceDependencyModel model, Str
         }
     }
 
+    public static String getJobExecuteResourceFilePath(JobResourceDependencyModel model, IRepositoryViewObject jobObject,
+            String jobLabel,
+            String resourceVersion) {
+        String path = null;
+        String extResPath = getProcessFolder(jobObject) + jobLabel.toLowerCase() + SRC_EXTRESOURCE_FOLDER;
```
Comment (zwxue) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/util/ResourceDependenciesUtil.java]: can use RunprocessService.getTalendJobJavaProject.getExtsourceFolder() here

File: main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/CodeGenerator.java (hunk 1 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.codegen/src/main/java/org/talend/designer/codegen/CodeGenerator.java]: Modified, thanks

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/util/ResourceDependenciesUtil.java (hunk 2 above)
Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/util/ResourceDependenciesUtil.java]: Modified, thanks

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/service/ResourcesDependenciesService.java [hunk 3]
```diff
@@ -89,12 +91,12 @@ public String getResourcePathForContext(IProcess process, String resourceContext
                     if (repoObject != null) {
                         JobResourceDependencyModel model = new JobResourceDependencyModel(
                                 (ResourceItem) repoObject.getProperty().getItem());
-                        StringBuffer joblabel = new StringBuffer();
-                        if (StringUtils.isNotBlank(property.getItem().getState().getPath())) {
-                            joblabel.append(property.getItem().getState().getPath() + "/");
+                        String jobLabel = JavaResourcesHelper.getJobFolderName(property.getLabel(), property.getVersion());
+                        if (ProcessorUtilities.isExportConfig()) {
```
Comment (zwxue) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/service/ResourcesDependenciesService.java]: if(!ProcessorUtilities.isExportConfig())  ?

Comment (jding-tlnd) [reply main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/editor/dependencies/service/ResourcesDependenciesService.java]: if(ProcessorUtilities.isExportConfig()) relativePath else fullPath

**Pull request merged**